packages/module-services/src/provider.ts

```diff
@@ -0,0 +1,43 @@
+import { IHttpClient } from '@equinor/fusion-framework-module-http';
+import { ClientMethod } from './lib/types';
+
+import { ApiClientFactory } from './types';
+import { ContextApiClient } from './lib/context';
+
+enum Services {
+    Context = 'context',
+}
+
+type ApiServices<TClient extends IHttpClient, TMethod extends keyof ClientMethod> = {
+    [Services.Context]: ContextApiClient<TClient, TMethod>;
+};
+
+export interface IApiProvider {
+    createApiClient<TService extends Services, TMethod extends keyof ClientMethod>(
+        name: TService,
+        version: TMethod
+    ): Promise<ApiServices<IHttpClient, TMethod>[TService]>;
+}
+
+type ApiProviderCtorArgs = {
+    createClient: ApiClientFactory;
+};
+
+export class ApiProvider implements IApiProvider {
+    protected __createClientFn: ApiClientFactory;
```

Comment:
hvorfor dobbel _ ? ser det ikke er konsekvent i koden. du bruker enkel noen plasser.

Reply:
type when changed from private to protected